### PR TITLE
adding comments to addUpdate for filter

### PR DESCRIPTION
**Pull request by mcamurri**

### CODE REVIEW

File: mav/mav-state-estimator/src/mav_state_est/mav_state_est.cpp [hunk 1]
```diff
@@ -27,40 +27,57 @@ MavStateEstimator::~MavStateEstimator()
...
   }
+  // If we don't want to process, we exit
   if (!roll_forward) {
     return;
   }
 
+  // We get the update before the first unprocessed update
   updateHistory::historyMapIterator prev_it = unprocessed_updates_start;
-  prev_it--;
+  --prev_it;
```
Comment (mauricefallon) [inline mav/mav-state-estimator/src/mav_state_est/mav_state_est.cpp]: arbitrary change of incrementor here with no comment to explain

Comment (mcamurri) [inline mav/mav-state-estimator/src/mav_state_est/mav_state_est.cpp]: Prefix increment for non primitive types is more efficient:
https://en.wikibooks.org/wiki/Optimizing_C%2B%2B/Writing_efficient_code/Constructions_and_destructions#Increment.2Fdecrement_operators

Comment (mauricefallon) [inline mav/mav-state-estimator/src/mav_state_est/mav_state_est.cpp]: this is a very minor improvement hidden within many added comments.

Comment (mcamurri) [inline mav/mav-state-estimator/src/mav_state_est/mav_state_est.cpp]: Should I create a PR only for that one?

**Pull request closed without merging**